src/Analyzers/CSharp/Analyzers/UseCollectionInitializer/CSharpUseCollectionInitializerDiagnosticAnalyzer.cs

```diff
@@ -24,6 +24,7 @@ internal sealed class CSharpUseCollectionInitializerDiagnosticAnalyzer :
         MemberAccessExpressionSyntax,
         InvocationExpressionSyntax,
         ExpressionStatementSyntax,
+        LocalDeclarationStatementSyntax,
```

Comment:
passing this around as a strongly typed type-argument so that the signatures are correct in each language.